src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,35 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LOGIN_LENGTH = 6;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        if (user.getAge() == null) {
+            throw new InvalidDataException("Missing information: User age");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Missing information: User login");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Missing information: User Password");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
```

Comment:
The error message 'this login already exists' should start with a capital letter for consistency and clarity.